### PR TITLE
feat: generate deps via custom plugin

### PR DESCRIPTION
**Pull request by orsagie**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Introduces scala plugins to generate a graph output for the snyk plugin to parse and turn into a tree. 
See sample outputs in test/fixtures. 

#### Any background context you want to provide?
This is introduced as a way to use less memory to generate a dependency tree from sbt.

### CODE REVIEW

File: test/functional/parse-sbt.test.ts [hunk 1]
```diff
@@ -25,24 +24,25 @@ test('parse `sbt dependencies` output: multi configuration', (t) => {
     'testApp',
     'package name');
   t.equal(depTree.version, '1.0.1', 'package version');
-  t.true(depTree.multiBuild, 'multi build flag set');
+  // t.true(depTree.multiBuild, 'multi build flag set');
+  // Where is multiBuild flag looked at?
```
Comment (lwywoo) [inline test/functional/parse-sbt.test.ts]: Did you manage to find out where this flag is set?

Comment (lili2311) [reply test/functional/parse-sbt.test.ts]: https://github.com/snyk/snyk-sbt-plugin/blob/94a8d945a376333d98564b7ba271682ec7927012/lib/parse-sbt.ts#L128 seems to be here?

File: package.json [hunk 2]
```diff
@@ -33,7 +33,13 @@
     "typescript": "3.4.5"
   },
   "dependencies": {
+    "@types/node": "6.14.6",
```
Comment (dkontorovskyy) [inline package.json]: should this be in `dependencies`? We already have it in `devDependencies`

File: package.json [hunk 3]
```diff
@@ -33,7 +33,13 @@
...
     "@types/sinon": "7.0.11",
+    "child_process": "1.0.2",
+    "fs": "0.0.1-security",
+    "path": "0.12.7",
+    "semver": "^6.1.2",
+    "tmp": "^0.1.0",
     "tree-kill": "^1.2.1",
     "tslib": "1.9.3"
```
Comment (dkontorovskyy) [inline package.json]: Shouldn't this also be in `devDependencies`?

**Pull request merged**